modules/tinymce/src/themes/silver/test/ts/browser/editor/EventsTest.ts

```diff
@@ -0,0 +1,75 @@
+import { Mouse, UiFinder, Waiter } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyContentActions, TinyDom, TinyHooks, TinyUiActions } from '@ephox/mcagar';
+import { SugarBody } from '@ephox/sugar';
+import { assert } from 'chai';
+
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.themes.silver.editor.EventsTest', () => {
+  const hook1 = TinyHooks.bddSetup({
+    base_url: '/project/tinymce/js/tinymce',
+    plugins: 'link',
+  }, [ Theme ]);
+
+  const hook2 = TinyHooks.bddSetup({
+    base_url: '/project/tinymce/js/tinymce',
+    plugins: 'link',
+  }, [ Theme ]);
+
+  const waitToOpen = (selector: string) => {
+    return Waiter.pTryUntil(`Wait ${selector} to open`, () => UiFinder.exists(SugarBody.body(), selector));
+  };
```

Comment:
```
  const pWaitToOpen = (selector: string) =>
    Waiter.pTryUntil(`Wait for a menu to open`, () => UiFinder.exists(SugarBody.body(), selector));
```
Shorthand and we don't need to include the selector as it'll be in the assertion error message. We also have a convention to prefix test functions that are async with `p` currently.

Reply:
Yeah, but if remove the selector from the message and pass another thing that's not a menu is not gonna sound good. Unless the method is `pWaitMenuToOpen` without params...

Reply:
That's all you're using it for though... unless this is moved elsewhere to be generic, then that's not something we need to worry about here. You're right it should be renamed though.